ovos_utils/json_helper.py

```diff
@@ -2,21 +2,35 @@
 from ovos_utils.parse import fuzzy_match
 
 
-def merge_dict(base, delta):
+def merge_dict(base, delta, merge_lists=False, skip_empty=False, no_dupes=True):
     """
         Recursively merging configuration dictionaries.
 
         Args:
             base:  Target for merge
             delta: Dictionary to merge into base
+            merge_lists: if a list is found merge contents instead of replacing
+            skip_empty: if an item in delta is empty, dont overwrite base
+            no_dupes: when merging lists deduplicate entries
     """
 
     for k, dv in delta.items():
         bv = base.get(k)
         if isinstance(dv, dict) and isinstance(bv, dict):
             merge_dict(bv, dv)
         else:
-            base[k] = dv
+            if skip_empty and not dv:
+                # dont replace if new entry is empty
+                pass
+            elif merge_lists and \
+                    isinstance(base[k], list) and \
+                    isinstance(dv, list):
```

Comment:
Consider using `all()`:

```python3
elif all(( #notice iterator
          merge_lists,
          isinstance(base[k], list),
          isinstance(dv, list) ))
```

this will stop on the first `False`.